### PR TITLE
add --all and --human cases into virsh domblkinfo

### PR DESCRIPTION
**Pull request by chunfuwen**
--all and --human are new options added to virsh domblkinfo
and related output format need to be checked

Signed-off-by: chunfuwen <chwen@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py [hunk 1]
```diff
@@ -16,34 +18,44 @@ def run(test, params, env):
...
-    def attach_disk_test():
+    def attach_disk_test(test_disk_source, front_dev):
         """
         Attach-disk testcase.
         1.Attch a disk to guest.
         2.Perform domblkinfo operation.
         3.Detach the disk.
 
+        :param test_disk_source disk source file path.
+        :param front_dev front end device name.
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py]: format error, need add ":" after parameters

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py [hunk 2]
```diff
@@ -57,11 +69,22 @@ def check_disk_info():
...
-            if disk_size != size and disk_size_check:
-                test.fail("Command domblkinfo output is wrong! "
-                          "'%d' != '%d'" % (disk_size, size))
+            if "--human" in extra and not any(re.findall(r'GiB|MiB', lines[0], re.IGNORECASE)):
+                test.fail("Command domblkinfo human output is wrong")
+            if "--all" in extra:
+                blocklist = vm_xml.VMXML.get_disk_blk(vm_name)
+                if not all(re.findall(r''.join(block), output_source, re.IGNORECASE) for block in blocklist):
+                    test.fail("Command domblkinfo --all output is wrong")
+                return
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py]: why return here? Please give more explanation for codes

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py [hunk 3]
```diff
@@ -104,11 +127,14 @@ def check_disk_info():
     elif vm_ref == "uuid":
         vm_ref = domuuid
 
+    if any(re.findall(r'--all|--human', extra, re.IGNORECASE)) and not libvirt_version.version_compare(4, 5, 0):
+        test.skip("--all and --human options are supported until libvirt 4.5.0 version")
+
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py]: test.cancel

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py [hunk 4]
```diff
@@ -16,34 +18,44 @@ def run(test, params, env):
...
+                disk_source = ""
+                vm_ref = "%s %s" % (vm_name, extra)
+            result_source = virsh.domblkinfo(vm_ref, disk_source,
                                              ignore_status=True, debug=True)
             status_source = result_source.exit_status
             output_source = result_source.stdout.strip()
             if driver == "qemu":
-                result_target = virsh.domblkinfo(vm_ref, front_dev,
+                if "--all" in extra:
+                    front_device = ""
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py]: why not merge it with above judgement for "--all"

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py (hunk 1 above)
Comment (chunfuwen) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py]: done

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py (hunk 4 above)
Comment (chunfuwen) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py]: condition is different,here driver == "qemu": , moreover, target_device is different ,too.

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py (hunk 2 above)
Comment (chunfuwen) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py]: no need further operation for this case

File: libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py (hunk 3 above)
Comment (chunfuwen) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_domblkinfo.py]: updated

**Pull request merged**